src/Symfony/Component/AssetMapper/ImportMap/ImportMapRenderer.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+/**
+ * @experimental
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class ImportMapRenderer
+{
+    public function __construct(
+        private readonly ImportMapManager $importMapManager,
+        private readonly string $charset = 'UTF-8',
+        private readonly string|false $polyfillUrl = ImportMapManager::POLYFILL_URL,
+        private readonly array $scriptAttributes = [],
+    ) {
+    }
+
+    public function render(?string $entryPoint = null): string
+    {
+        $attributeString = '';
+        foreach ($this->scriptAttributes as $name => $value) {
```

Comment:
the `type` name should probably be forbidden. And for safety, we should probably also forbid `src`